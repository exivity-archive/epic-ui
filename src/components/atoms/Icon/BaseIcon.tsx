import * as React from 'react'

import styled from "styled-components";

interface IBaseIconProps {
  className?: string
  extraClassName?: string
  onClick?: (e: React.MouseEvent<HTMLSpanElement, MouseEvent>) => void
  children?: JSX.Element
}
 
const BaseIcon: React.FC<IBaseIconProps> = ({ className, extraClassName, children, onClick }) => {
  return <span className={className + ' ' + extraClassName} onMouseDown={onClick}>{children}</span>
}

export default styled(BaseIcon)`
  cursor: auto;
  font-family: ${props => `Material Icons, ${props.theme.global.font}`};
  font-weight: normal;
  font-style: normal;
  font-size: 24px;

  line-height: 1;
  vertical-align: middle;

  text-transform: none;
  letter-spacing: normal;
  word-wrap: normal;
  white-space: nowrap;
  direction: ltr;

  /* Support for all WebKit browsers. */
  -webkit-font-smoothing: antialiased;

  /* Support for Safari and Chrome. */
  text-rendering: optimizeLegibility;

  /* Support for Firefox. */
  -moz-osx-font-smoothing: grayscale;

  /* Support for IE. */
  font-feature-settings: 'liga';
 
`
