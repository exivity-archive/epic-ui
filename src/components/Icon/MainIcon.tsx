import * as React from 'react'
import styled, { css } from 'styled-components';
import BaseIcon from './BaseIcon';

interface IMainIconProps {
  subIcon?: string
  children: string
  className?: string
  onClick?: (e: React.MouseEvent<HTMLSpanElement, MouseEvent>) => void
}

const MainIcon: React.FC<IMainIconProps> = ({ children, className, onClick }) => {
  return <BaseIcon extraClassName={className} onClick={onClick}>{children}</BaseIcon>
}

export default styled(MainIcon)`
  ${props => props.subIcon && css`
    /* Zero out the line-height so that it doesn't
      interfere with the positioning that follows */
      line-height: 0;

      /* Where the magic happens: makes all browsers position
        the sup/sup properly, relative to the surrounding text */
      position: relative;

      /* Note that if you're using Eric Meyer's reset.css, this
        is already set and you can remove this rule */
      vertical-align: baseline;

      /* Move the superscripted text up */
      top: -.2em;

      font-size: 16px;
  `}
  ${props => !props.children && props.subIcon && css`
      width: 16px;
      height: 16px;
  `}
`

