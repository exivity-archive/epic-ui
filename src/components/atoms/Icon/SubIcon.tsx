import * as React from 'react'
import styled from 'styled-components';
import BaseIcon from './BaseIcon';

interface ISubIconProps {
  children: JSX.Element
  onClick?: (e: React.MouseEvent<HTMLSpanElement, MouseEvent>) => void
  className?: string
}

const SubIcon: React.FC<ISubIconProps> = ({ children, onClick, className }) => {
  return <BaseIcon onClick={onClick} extraClassName={className}>{children}</BaseIcon>
}

export default styled(SubIcon)`
  /* Zero out the line-height so that it doesn't
    interfere with the positioning that follows */
  line-height: 0;

  /* Where the magic happens: makes all browsers position
    the sup/sup properly, relative to the surrounding text */
  position: relative;

  /* Note that if you're using Eric Meyer's reset.css, this
    is already set and you can remove this rule */
  vertical-align: baseline;

  /* Move the subscripted text down, but only
       half as far down as the superscript moved up */
  bottom: -.3em;
  margin-left: -.4em;

  font-size: 10px;

`