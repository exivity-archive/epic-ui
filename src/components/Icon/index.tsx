import React, { Fragment } from 'react'
import MainIcon from './MainIcon';
import SubIcon from './SubIcon';

interface IIconProps {
  children: string
  subIcon?: string
  onClick?: (event: React.MouseEvent<HTMLSpanElement, MouseEvent>) => void
}

const Icon: React.FC<IIconProps> = ({ children, subIcon, onClick }) => (
  <Fragment>
    <MainIcon onClick={onClick} subIcon={subIcon}>{children}</MainIcon>
    {!!subIcon && <SubIcon onClick={onClick}>{subIcon}</SubIcon>}
  </Fragment>
)

export default Icon
