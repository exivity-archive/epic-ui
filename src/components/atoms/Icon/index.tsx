import React, { Fragment } from 'react'
import MainIcon from './MainIcon';
import SubIcon from './SubIcon';

interface IIconProps {
  icon: JSX.Element
  subIcon?: JSX.Element
  onClick?: (event: React.MouseEvent<HTMLSpanElement, MouseEvent>) => void
}

const Icon: React.FC<IIconProps> = ({ icon, subIcon, onClick}) => (
  <Fragment>
    <MainIcon onClick={onClick} icon={icon} subIcon={subIcon} />
    {!!subIcon && <SubIcon onClick={onClick} icon={subIcon} />}
  </Fragment>
)

export default Icon
