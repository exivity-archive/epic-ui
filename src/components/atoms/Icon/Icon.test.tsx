import React from 'react'
import { shallowWithTheme } from '../../../utils/testing/shallowWithTheme'
import { MdAccountBalanceWallet } from 'react-icons/md'

import Icon from './'

test('renders icon', () => {
  const button = shallowWithTheme(<Icon icon={<MdAccountBalanceWallet/>} onClick={() => { return }} />)
  expect(button).toMatchSnapshot()
})

test('renders icon with subicon', () => {
  const button = shallowWithTheme(<Icon icon={<MdAccountBalanceWallet/>} subIcon={<MdAccountBalanceWallet/>} onClick={() => { return }} />)
  expect(button).toMatchSnapshot()
})
