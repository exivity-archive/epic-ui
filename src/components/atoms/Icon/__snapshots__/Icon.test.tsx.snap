// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders icon 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "gray": "#aaa",
        },
        "global": Object {
          "fontFamily": "-apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, sans-serif !default;",
          "margin": "20px !default",
        },
      }
    }
  >
    <Icon
      icon={<MdAccountBalanceWallet />}
      onClick={[Function]}
    />
  </ThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": undefined,
        "_currentValue2": undefined,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": undefined,
          "_currentValue2": undefined,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`renders icon with subicon 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "gray": "#aaa",
        },
        "global": Object {
          "fontFamily": "-apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, sans-serif !default;",
          "margin": "20px !default",
        },
      }
    }
  >
    <Icon
      icon={<MdAccountBalanceWallet />}
      onClick={[Function]}
      subIcon={<MdAccountBalanceWallet />}
    />
  </ThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": undefined,
        "_currentValue2": undefined,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": undefined,
          "_currentValue2": undefined,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
